K2Field.K2NE.ServiceBroker/ServiceObjects/ExcelServices/ExcelImportServiceSO.cs

```diff
@@ -0,0 +1,528 @@
+                        {
+                            relationshipId = sheet.Id.Value;
+                        }
+                        else
+                        {
+                            throw new ApplicationException(string.Format(Resources.ExcelImportSheetNotExist, sheetName));
+                        }
+                    }
+
+                    // Get sheet1 Part of Spread Sheet Document
+                    WorksheetPart worksheetPart = (WorksheetPart)spreadsheetDocument.WorkbookPart.GetPartById(relationshipId);
+
+                    // Get Data in Excel file
+                    SheetData sheetData = worksheetPart.Worksheet.Elements<SheetData>().First();
+                    IEnumerable<Row> rowcollection = sheetData.Descendants<Row>();
+
+                    if (rowcollection.Count() == 0)
+                    {
+                        return dt;
+                    }
+
+                    // Add columns (identified by a conventional header row)
+                    foreach (Cell cell in rowcollection.ElementAt(0))
+                    {
+                        dt.Columns.Add(GetValueOfCell(spreadsheetDocument, cell));
+                    }
+
+                    // Add rows into DataTable
+                    foreach (Row row in rowcollection)
+                    {
+                        DataRow temprow = dt.NewRow();
+                        int currentColumnIndex = 0;
+                        for (int i = 0; i < row.Descendants<Cell>().Count(); i++)
+                        {
+                            //Excel does not create an element for empty cells. This can lead to apparently "offset" data in the row. See http://stackoverflow.com/questions/3837981/reading-excel-open-xml-is-ignoring-blank-cells
+                            //Overcome this with some refe
+                            int cellColumnIndex = (int)GetColumnIndexFromRefName(GetColumnRefName(row.Descendants<Cell>().ElementAt(i).CellReference));
+                            while (currentColumnIndex < cellColumnIndex)
+                            {   //then we need to recreate the blank that was skipped over by Excel's efficient non-storage of empties
+                                temprow[currentColumnIndex] = string.Empty;
+                                currentColumnIndex++;
+                            }
+
+                            temprow[currentColumnIndex] = GetValueOfCell(spreadsheetDocument, row.Descendants<Cell>().ElementAt(i));
+                            currentColumnIndex++;
+                        }
+
+                        // Add the row to DataTable
+                        // note the rows include header row
+                        if (!AreAllColumnsEmpty(temprow))
+                            dt.Rows.Add(temprow);
+                    }
+                }
+
+                // Here remove header row
+                dt.Rows.RemoveAt(0);
+                return dt;
+            }
+            catch (IOException ex)
```

Comment:
This try/catch block litterly does nothing. The optimizer will remove it completely.

Reply:
Removed.